editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -1018,18 +1018,19 @@ void SpriteFramesEditor::_update_library(bool p_skip_selector) {
 	}
 
 	for (int i = 0; i < frames->get_frame_count(edited_anim); i++) {
-		String name;
+		String name = itos(i);
 		Ref<Texture2D> texture = frames->get_frame_texture(edited_anim, i);
 		float duration = frames->get_frame_duration(edited_anim, i);
-		String duration_string;
-		if (duration != 1.0f) {
-			duration_string = String::utf8(" [ ×") + String::num_real(frames->get_frame_duration(edited_anim, i)) + " ]";
-		}
 
 		if (texture.is_null()) {
-			name = itos(i) + ": " + TTR("(empty)") + duration_string;
-		} else {
-			name = itos(i) + ": " + texture->get_name() + duration_string;
+			texture = empty_icon;
+			name += ": " + TTR("(empty)");
+		} else if (!texture->get_name().is_empty()) {
+			name += ": " + texture->get_name();
+		}
+
+		if (duration != 1.0f) {
+			name += String::utf8(" [ × ") + String::num(duration, 2) + " ]";
```

Comment:
It is better to not have the space in the brackets here.

So `[ x 4.2 ]` becomes `[x4.2]`

I can accept an argument for a space after the `x`, but a space to delimit the brackets is absolutely superfluous as they are themselves delimiters.

Reply:
You're right. I added a space after the `×` and removed the decimals because this is what it looks like in the Frame Duration field:

![](https://user-images.githubusercontent.com/47700418/212757880-2c0e0e3f-2a04-4ac1-af07-1d6fe879d3fb.png)

I also had an idea to remove the brackets (something like `#0 × 4`), but I didn't like how it looked.